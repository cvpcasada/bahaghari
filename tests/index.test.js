import { clearInterval } from 'timers';

const Bahaghari = require(`../src/index`);

describe(`Razer interface`, () => {
  it(`connects successfully`, async () => {
    const chromaInstance = await Bahaghari.createChroma({
      title: 'Razer Chroma SDK RESTful Test Application',
      description: 'This is a REST interface test application',
      author: {
        name: 'Chroma Developer',
        contact: 'www.razerzone.com',
      },
      device_supported: [
        'keyboard',
        'mouse',
        'headset',
        'mousepad',
        'keypad',
        'chromalink',
      ],
      category: 'application',
    });

    // close the heartbeat
    clearInterval(chromaInstance.heartbeat);

    expect(chromaInstance.application).toMatchSnapshot();
    expect(chromaInstance.sessionid).toBeDefined();
    expect(chromaInstance.uri).toBeDefined();
  });
});
