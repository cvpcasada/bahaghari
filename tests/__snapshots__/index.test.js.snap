// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Razer interface connects successfully 1`] = `
Object {
  "author": Object {
    "contact": "www.razerzone.com",
    "name": "Chroma Developer",
  },
  "category": "application",
  "description": "This is a REST interface test application",
  "device_supported": Array [
    "keyboard",
    "mouse",
    "headset",
    "mousepad",
    "keypad",
    "chromalink",
  ],
  "title": "Razer Chroma SDK RESTful Test Application",
}
`;
